src/main/java/io/pravega/perf/WriterWorker.java

```diff
@@ -5,47 +5,64 @@
  * you may not use this file except in compliance with the License.
  * You may obtain a copy of the License at
  *
- *     http://www.apache.org/licenses/LICENSE-2.0
+ *      http://www.apache.org/licenses/LICENSE-2.0
  */
 
 package io.pravega.perf;
 
+import javax.annotation.concurrent.NotThreadSafe;
 import java.io.IOException;
-import java.time.Duration;
-import java.time.Instant;
+import java.nio.charset.StandardCharsets;
 import java.util.Random;
 import java.util.concurrent.Callable;
-import java.util.concurrent.CompletableFuture;
 import java.util.concurrent.ExecutionException;
 
 /**
- * abstract class for Writers.
+ * Abstract class for Writers.
  */
 public abstract class WriterWorker extends Worker implements Callable<Void> {
+    final private static int MS_PER_SEC = 1000;
     final private Performance perf;
-    final private ThroughputController tput;
+    final private String payload;
+    final private int eventsPerSec;
 
     WriterWorker(int sensorId, int events, int secondsToRun,
-                 boolean isRandomKey, int messageSize, Instant start,
-                 PerfStats stats, String streamName, ThroughputController tput) {
+                 boolean isRandomKey, int messageSize, long start,
+                 PerfStats stats, String streamName, int eventsPerSec, boolean writeNread) {
```

Comment:
`writeNread` -> means `writeAndRead`, right? Maybe `writeAndRead` would be easier to understand. Otherwise, it can be confused to write `N` events, and then read.

Reply:
yes , its renamed as  writeAndRead